extensions/amp-content-viewer/0.1/amp-content-viewer.js

```diff
@@ -0,0 +1,696 @@
+    return layout == Layout.FIXED || Layout.FILL;
+  }
+
+  /** @override */
+  isRelayoutNeeded() {
+    return true;
+  }
+
+  /**
+   * Returns the boundaries of the content element.
+   * @return {?../../../src/layout-rect.LayoutRectDef}
+   */
+  getContentBox() {
+    return this.contentBox_;
+  }
+
+  /**
+   * Returns the content element.
+   * @return {?Element}
+   */
+  getContent() {
+    return this.content_;
+  }
+
+  /**
+   * Returns the boundaries of the content element with the offset if it was
+   * moved by a gesture.
+   * @return {?../../../src/layout-rect.LayoutRectDef}
+   */
+  getContentBoxWithOffset() {
+    if (this.posX_ == 0 && this.posY_ == 0 || !this.contentBox_) {
+      return this.contentBox_;
+    }
+    const expansionScale = (this.scale_ - 1) / 2;
+    return moveLayoutRect(
+        expandLayoutRect(this.contentBox_, expansionScale, expansionScale),
+        this.posX_,
+        this.posY_
+    );
+  }
+
+  /**
+   * Checks to see if an element is supported.
+   * @param {Element} element
+   * @return {boolean}
+   * @private
+   */
+  elementIsSupported_(element) {
+    return ELIGIBLE_TAGS[element.tagName.toLowerCase()];
+  }
+
+  /**
+   * Measures the content viewer and content sizes and positioning.
+   * This must be called AFTER the source element has already been
+   * laid out.
+   * @private
+   */
+  measure_() {
+    this.elementBox_ = layoutRectFromDomRect(this.element
+        ./*OK*/getBoundingClientRect());
+
+    const sourceAspectRatio = this.sourceWidth_ / this.sourceHeight_;
+    let height = Math.min(this.elementBox_.width / sourceAspectRatio,
+        this.elementBox_.height);
+    let width = Math.min(this.elementBox_.height * sourceAspectRatio,
+        this.elementBox_.width);
+
+    if (Math.abs(width - this.sourceWidth_) <= 16
+    && Math.abs(height - this.sourceHeight_ <= 16)) {
+      width = this.sourceWidth_;
+      height = this.sourceHeight_;
+    }
+
+    this.contentBox_ = layoutRectLtwh(
+        Math.round((this.elementBox_.width - width) / 2),
+        Math.round((this.elementBox_.height - height) / 2),
+        Math.round(width),
+        Math.round(height));
+
+    // Adjust max scale to at least fit the screen.
+    const elementBoxRatio = this.elementBox_.width /
+     this.elementBox_.height;
+    const maxScale = Math.max(
+        elementBoxRatio / sourceAspectRatio,
+        sourceAspectRatio / elementBoxRatio
+    );
+    this.maxScale_ = Math.max(DEFAULT_MAX_SCALE, maxScale);
+
+    // Reset zoom and pan.
+    this.startScale_ = this.scale_ = 1;
+    this.startX_ = this.posX_ = 0;
+    this.startY_ = this.posY_ = 0;
+    this.updatePanZoomBounds_(this.scale_);
+  }
+
+  /**
+   * Measures and resets the content dimensions, after the element
+   * dimensions changes.
+   * @return {!Promise}
+   */
+  resetContentDimensions_() {
+    return this.measureElement(() => this.measure_()).then(() => {
+      const content = dev().assertElement(this.content_);
```

Comment:
It can't because the mutate applies explicitly to the `content` only, and `measure` applies to the entire thing. Basically you're measuring the `amp-content-viewer`, but only mutating the `content` inside the `amp-content-viewer`.